crates/net/eth-wire/src/multiplex.rs

```diff
@@ -312,13 +312,14 @@ impl ProtocolProxy {
             return Err(io::ErrorKind::InvalidInput.into())
         }
 
-        let mut masked_bytes = BytesMut::zeroed(msg.len());
-        masked_bytes[0] = msg[0]
-            .checked_add(self.shared_cap.relative_message_id_offset())
-            .ok_or(io::ErrorKind::InvalidInput)?;
+        let offset = self.shared_cap.relative_message_id_offset();
+        if offset == 0 {
```

Comment:
for the eth protocol this only happens once when the initial status message is sent.

so likely not worth checking